implementation/src/main/java/io/smallrye/mutiny/groups/UniOnNotNull.java

```diff
@@ -198,4 +200,29 @@ public UniOnNotNull(Uni<T> upstream) {
         });
     }
 
+    /**
+     * If the current {@link Uni} contains a item, the produced {@link Uni} emits a failure produced
+     * using the given {@link Supplier}.
+     *
+     * @param supplier the supplier to produce the failure, must not be {@code null}, must not produce {@code null}
+     * @return the new {@link Uni}
+     */
+    public Uni<T> failWith(Supplier<? extends Throwable> supplier) {
+        Supplier<? extends Throwable> actual = Infrastructure.decorate(nonNull(supplier, "supplier"));
+        return Infrastructure.onUniCreation(transformToUni(ignored -> {
+            Throwable failure = Objects.requireNonNull(actual.get(), SUPPLIER_PRODUCED_NULL);
```

Comment:
You can use `ParameterValidation` from Mutiny instead

Reply:
Done!